client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerops
+
+import (
+	"context"
+	log "github.com/Sirupsen/logrus"
+	dockerClient "github.com/docker/engine-api/client"
+	dockerTypes "github.com/docker/engine-api/types"
+)
+
+const (
+	// dockerAPIVersion: docker engine 1.24 and above support this api version
+	dockerAPIVersion = "v1.24"
+	// dockerUSocket: Unix socket on which Docker engine is listening
+	dockerUSocket = "unix:///var/run/docker.sock"
+)
+
+// DockerOps is the interface for docker host related operations
+type DockerOps struct {
+	Dockerd *dockerClient.Client
+}
+
+func NewDockerOps() *DockerOps {
+	var d *DockerOps
+
+	client, err := dockerClient.NewClient(dockerUSocket, dockerAPIVersion, nil, nil)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"error": err},
+		).Error("Failed to create client for Docker ")
+		return nil
+	}
+
+	d = &DockerOps{
+		Dockerd: client,
+	}
+
+	return d
+}
+
+// VolumeCreate - create volume from docker host with specific volume driver
+func (d *DockerOps) VolumeCreate(volumeDriver string, volName string, options map[string]string) error {
+	dockerVolOptions := dockerTypes.VolumeCreateRequest{
+		Driver:     volumeDriver,
+		Name:       volName,
+		DriverOpts: options,
+	}
+
+	_, err := d.Dockerd.VolumeCreate(context.Background(), dockerVolOptions)
+
+	return err
+}
+
+// VolumeCreate - remove volume from docker host with specific volume driver
+func (d *DockerOps) VolumeRemove(volName string) error {
+	return d.Dockerd.VolumeRemove(context.Background(), volName)
+}
+
+// StartSMBServer - Start SMB server
+func (d *DockerOps) StartSMBServer(volName string) bool {
```

Comment:
Inside this function, we will talk to docker host in order to start a swarm service. It will need the same information as other operations who need to talk to docker host. That's why I put it here. All the functions inside this structure is related to docker host.